src/main/java/com/google/firebase/database/snapshot/NodeUtilities.java

```diff
@@ -58,6 +60,10 @@ public static Node NodeFromJSON(Object value, Node priority) throws DatabaseExce
         return new DoubleNode((Double) value, priority);
       } else if (value instanceof Boolean) {
         return new BooleanNode((Boolean) value, priority);
+      } else if (value instanceof BigDecimal) {
+        return new DoubleNode(((BigDecimal) value).doubleValue(), priority);
+      } else if (value instanceof BigInteger) {
+        return new LongNode(((BigInteger)value).longValue(), priority);
```

Comment:
Same question applies here. What happens when the BigInteger is too large to fit in a long?

Reply:
Ditto for BigInteger -- we should serialize these to String to maintain the full representation.